### ISSUE TITLE
Feature Request: --dry-run

### ISSUE THREAD

**Issue by Miserlou** [COMBINE-lab/salmon#189]
Hello!

We are currently using Salmon as part of a very large data processing system. We need to continuously test the end-to-end functionality of our system, which includes a Salmon execution. Unfortunately, this increases our build time from 5 minutes to an hour and a half.

It'd be really, really excellent if we could have a way to _simulate_ a Salmon run to verify the correct installation of Salmon and the Salmon-compatible collection of our input data without actually having to do the full Salmon run.

Is there any chance you'd consider adding a `--dry-run` argument to run Salmon without actually running the big crunch? Ideally this could also output some dummy results files, but it doesn't have to if you don't like that.

Thoughts?

Related: https://github.com/data-refinery/data-refinery/issues/106

**Comment by mdshw5** [comment]
What's wrong with actually running salmon, monitoring stderr for what you need, and then killing the process?

**Comment by Miserlou** [comment]
I mean, that's pretty hideous and unusual. =/ It's also not exactly what we want, since the steps of the internal processes may have different requirements.

Having `--dry-run` is pretty common for UNIX tools, for instance `rsync`.

**Comment by mdshw5** [comment]
I don't completely disagree, and don't speak for this project as anything other than a user, but `until 'salmon [args]' | grep -m 1 "Starting optimizer"; do : ; done` seems pretty unix-y to me.

**Comment by rob-p** [comment]
Hi @Miserlou,

  I'm not necessarily opposed to this.  What exactly would the dry-run do?  For example, would it simply check if the input files exist, try to load the index, etc.?  This seems like it could be useful functionality, though, in my experience `--dry-run` commands usually aren't effectful (i.e. they usually don't create directories or output files).

--Rob

**Comment by Miserlou** [comment]
Yes, just make sure the input files exist and are in the right format, and, to the extent possible, will cause a successful execution. I wouldn't expect a dry run to cause any changes to the file system, just to fail for bad input.
